### ISSUE TITLE
CRA unit tests are polluted with warnings for PropTypes and CreateClass

### ISSUE THREAD

**Issue by JacobRodriguezSSI** [erikras/redux-form#3411]
After installing redux-form and converting a widely used component, I'm suddenly seeing lots of deprecation warnings for React.PropTypes and React.CreateClass (a pair for each parent component my reduxForm component is used by). It's not terrible as tests still pass, but it adds noise to the test results.

### To reproduce:
1. Install create-react-app v1.4.0
2. Run `create-react-app rf-warn-test`.
3. CD into `./rf-warn-test` then `npm install -S react-redux redux redux-form`
4. Run `npm test` and leave open.
* Notice that the test will pass and no warnings or errors
5. Modify App.js to use `redux-form`:
```
import React, { Component } from 'react';
import {reduxForm} from 'redux-form';

import logo from './logo.svg';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to React</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export default reduxForm({
  form: 'AppForm'
})(App);
```
Now, notice that the unit test shows 2 warnings:
```
console.warn node_modules/react/lib/lowPriorityWarning.js:40
    Warning: Accessing PropTypes via the main React package is deprecated, and will be removed in  React v16.0. Use the latest available v15.* prop-types package from npm instead. For info on usage, compatibility, migration and more, see https://fb.me/prop-types-docs

console.warn node_modules/react/lib/lowPriorityWarning.js:40
    Warning: Accessing createClass via the main React package is deprecated, and will be removed in React v16.0. Use a plain JavaScript class instead. If you're not yet ready to migrate, create-react-class v15.* is available on npm as a temporary, drop-in replacement. For more info see https://fb.me/react-create-class
```
There are other issues (like store is not created or connected, etc), but in my production app, those things are working. The point is these 2 warnings get duplicated for every component that depends on a component connected with ReduxForm.

### Environment and Dependencies:
- OSX v10.11.6
- react: 15.6.1
- react-dom: 15.6.1
- react-redux: 5.0.6
- redux: 3.7.2
- redux-form: 7.0.4
- react-scripts: 1.0.13

### Other information
I know that in PR #2800 the issue was resolved, but it seems to be showing up only in the unit tests now and is just a little noisy. I assume that before #2800 was merged into 6.6.3, this was showing up in the browser console and `npm run`, just like it did in my own app before I migrated to `prop-types`.

I did a search for "PropTypes" in this repo and didn't find any "React.PropTypes" so I'm not sure where these warnings are coming from. I also checked the 2 dependencies that download with `redux-form`: `hoist-non-react-statics` had nothing but `propTypes:true`. `es6-error` didn't show anything at all (as I would have guessed).

**Comment by gustavohenke** [comment]
Duplicate of #3385

**Comment by erikras** [comment]
Published fix in [`v7.1.0`](https://github.com/erikras/redux-form/releases/tag/v7.1.0).
